catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -31,135 +36,71 @@
-                throw new PluginExecutionException(getExceptionMessage(CreateResponse.class.getSimpleName(),
-                        null,
-                        errors,
-                        OPERATION.CREATE));
-            }
-        }
+        execute(() -> create(input.getCreatedMetacards()));
         return input;
+
     }
 
     /**
      * Backs up updated metacards to the file system backup.
      *
      * @param input the {@link UpdateResponse} to process
-     * @return
-     * @throws PluginExecutionException
+     * @return {@link UpdateResponse}
      */
     @Override
-    public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
-        if (enableBackupPlugin && Requests.isLocal(input.getRequest())) {
-            LOGGER.debug("Updating metacards contained in UpdateResponse in backup.");
+    public UpdateResponse process(UpdateResponse input) {
 
-            if (rootBackupDir == null) {
-                throw new PluginExecutionException("No root backup directory configured.");
-            }
-
-            List<String> deleteErrors = new ArrayList<String>();
-            List<String> backupErrors = new ArrayList<String>();
-
-            List<Update> updates = input.getUpdatedMetacards();
-
-            for (Update update : updates) {
-                try {
-                    deleteMetacard(update.getOldMetacard());
-                } catch (IOException e) {
-                    deleteErrors.add(update.getOldMetacard()
-                            .getId());
-                }
-
-                try {
-                    backupMetacard(update.getNewMetacard());
-                } catch (IOException e) {
-                    backupErrors.add(update.getNewMetacard()
-                            .getId());
-                }
-            }
-
-            String exceptionMessage = null;
-
-            if (deleteErrors.size() > 0) {
-                exceptionMessage = getExceptionMessage(UpdateResponse.class.getSimpleName(),
-                        exceptionMessage,
-                        deleteErrors,
-                        OPERATION.DELETE);
-            }
-
-            if (backupErrors.size() > 0) {
-                exceptionMessage = getExceptionMessage(UpdateResponse.class.getSimpleName(),
-                        exceptionMessage,
-                        backupErrors,
-                        OPERATION.CREATE);
-            }
+        //Extract the old and new cards from Update objects
+        Function<Function<Update, Metacard>, List<Metacard>> getCards =
+                (fun) -> input.getUpdatedMetacards()
+                        .stream()
+                        .collect(Collectors.mapping(fun, Collectors.toList()));
 
-            if (deleteErrors.size() > 0 || backupErrors.size() > 0) {
-                throw new PluginExecutionException(exceptionMessage);
-            }
-        }
+        execute(() -> delete(getCards.apply(Update::getOldMetacard)));
```

Comment:
If I understand this correctly, we will be traversing the input metacards twice, right? Can we do something like http://www.leveluplunch.com/java/examples/partition-list/#java8 instead and iterate over the list only once?